exporter/collector/metrics.go

```diff
@@ -1184,14 +1217,19 @@ func metricPointValueType(pt pmetric.NumberDataPointValueType) metricpb.MetricDe
 	}
 }
 
-func mapMetricPointKind(m pmetric.Metric) (metricpb.MetricDescriptor_MetricKind, metricpb.MetricDescriptor_ValueType) {
+func (me *metricMapper) mapMetricPointKind(m pmetric.Metric) (metricpb.MetricDescriptor_MetricKind, metricpb.MetricDescriptor_ValueType) {
 	var kind metricpb.MetricDescriptor_MetricKind
 	var typ metricpb.MetricDescriptor_ValueType
 	switch m.Type() {
 	case pmetric.MetricTypeGauge:
 		kind = metricpb.MetricDescriptor_GAUGE
 		if m.Gauge().DataPoints().Len() > 0 {
-			typ = metricPointValueType(m.Gauge().DataPoints().At(0).ValueType())
+			_, supportedType := me.convertMetricKindToBoolIfSupported(m.Gauge().DataPoints().At(0), kind, m.Unit())
+			if supportedType != metricpb.MetricDescriptor_VALUE_TYPE_UNSPECIFIED {
+				typ = supportedType
+			} else {
+				typ = metricPointValueType(m.Gauge().DataPoints().At(0).ValueType())
+			}
```

Comment:
But then we would have to pass all those extra parameters - kind and and unit to `metricPointValueType` as well. 
Currently this method seems simple to me and serves a single purpose - convert `pmetric.NumberDataPointValueType` to its equivalent cloud monitoring type. I can see it going either way if you feel strongly about this.

Reply:
Just removed the support for Bool metrics from non-monotonic sum (which was interpreted as GUAGE) as per the spec we have. 

So now moving the check to `metricPointValueType ` would only simplify switch statement for a single case and I feel it may not increase readability by a lot.